dateutil/test/test_parser.py

```diff
@@ -238,6 +238,60 @@ def test_parse_bytearray(self):
         assert res == expected
 
 
+class TestTzinfoInputTypes(object):
+    def test_tzinfo_dict_could_return_none(self):
+        dstr = "2017-02-03 12:40 BRST"
+        result = parse(dstr, tzinfos={"BRST": None})
+        expected = datetime(2017, 2, 3, 12, 40)
+        assert result == expected
+
+    def test_tzinfos_callable_could_return_none(self):
+        dstr = "2017-02-03 12:40 BRST"
+        result = parse(dstr, tzinfos=lambda *args: None)
+        expected = datetime(2017, 2, 3, 12, 40)
+        assert result == expected
+
+    def test_invalid_tzinfo_input(self):
+        dstr = "2014 January 19 09:00 UTC"
+        # Pass an absurd tzinfos object
+        tzinfos = {"UTC": ValueError}
+        with pytest.raises(TypeError):
+            parse(dstr, tzinfos=tzinfos)
+
+    def test_valid_tzinfo_tzinfo_input(self):
+        dstr = "2014 January 19 09:00 UTC"
+        tzinfos = {"UTC": tz.tzutc()}
+        expected = datetime(2014, 1, 19, 9, tzinfo=tz.tzutc())
+        res = parse(dstr, tzinfos=tzinfos)
+        assert res == expected
+
+    def test_valid_tzinfo_unicode_input(self):
+        dstr = "2014 January 19 09:00 UTC"
+        tzinfos = {u"UTC": u"UTC+0"}
+        expected = datetime(2014, 1, 19, 9, tzinfo=tz.tzutc())
```

Comment:
Same comment as below.

Also, I realize that these tests would still succeed regardless of whether or not we're even *using* the `tzinfos` parameter. That makes me think that we should either be using `res.tzinfo is expected.tzinfo` or we should be using some none-UTC value here, or both.

One option is to have two tests - one for "UTC" as the offset specifically, that tests `is`, and one for something more general like `"LXQ"` where we are unlikely to accidentally hit the correct answer via another code path.